WordPressAuthenticator/NUX/NUXButton.swift

```diff
@@ -1,54 +1,285 @@
 import UIKit
 import WordPressShared
+import WordPressUI
+
+/// A protocol for an element that can display a UIActivityIndicatorView
+@objc public protocol ActivityIndicatorButton {
+    func showActivityIndicator(_ show: Bool)
+}
 
 /// A stylized button used by Login controllers. It also can display a `UIActivityIndicatorView`.
-@objc open class NUXButton: NUXSubmitButton {
-    // MARK: - Configuration
-    fileprivate let horizontalInset: CGFloat = 20
-    fileprivate let verticalInset: CGFloat = 12
-    fileprivate let maxFontSize: CGFloat = 22
+@objc open class NUXButton: UIButton, ActivityIndicatorButton {
+    @objc var isAnimating: Bool {
+        get {
+            return activityIndicator.isAnimating
+        }
+    }
+
+    @objc let activityIndicator: UIActivityIndicatorView = {
+        let indicator = UIActivityIndicatorView(activityIndicatorStyle: .white)
+        indicator.hidesWhenStopped = true
+        return indicator
+    }()
+
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+
+        if activityIndicator.isAnimating {
+            titleLabel?.frame = CGRect.zero
 
-    /// Configure the appearance of the button.
+            var frm = activityIndicator.frame
+            frm.origin.x = (frame.width - frm.width) / 2.0
+            frm.origin.y = (frame.height - frm.height) / 2.0
+            activityIndicator.frame = frm
```

Comment:
```
activityIndicator.frame = frm.integral
```

**Reason:** floating positions don't really exist (the screen requires integer coordinates). Anything that has a comma value results in a blurry view.

cc @mindgraffiti

Reply:
This is confusing. `x`, `y`, and `frame` all accept `CGFloat` values. Why do they make it like that if they really want `int`s?

Reply:
@mindgraffiti actually, my explanation was wrong. Lemme rephrase: positions are expressed in terms of Points, not Pixels. The problem is:

- **Non Retina Devices:** PositionX = 0.5 maps to `half a pixel in the X axis`
- **@2x Devices:** PositionX = 0.5 maps to `one pixel`
- **@3x Devices:** Position X = 0.5 maps to `one pixel and a half
 
Any "Pixel Position" that's not integral, when rendered, gets antialias applied. Which is why you usually wanna avoid non integral positions. (Exception is: when you're actually calculating things, so it maps to integral pixels).

[A little bit more about this here](https://stackoverflow.com/questions/14190807/why-are-uiview-frames-made-up-of-floats).